### PR TITLE
[PEPPER-1311] Implement ParticipantDataFixupService for AT genomic IDs

### PR DESCRIPTION
**Pull request by denniscunningham**
## Context
PEPPER-1311

This PR uses the AdminOperation facility to enable programmatic deletion of unneeded AT ParticipantData records. In this case the target is additional genomic ID and exit status records created for AT participants, which are causing confusion for AT study staff.

This particular service, `ParticipantDataFixupService`, can be extended in the future to handle other types of ParticipantData fixups. 

## Checklist

- [x] I have labeled the type of changes involved using the `C-*` labels.
- [x] I have assessed potential risks and labeled using the `R-*` labels.
- [x] I have considered error handling and alerts, and added `L-*` labels as needed.
- [ ] I have considered security and privacy, and added `I-*` labels as needed
- [ ] I have analyzed my changes for stability, fault tolerance, graceful degradation, performance bottlenecks and written a brief summary in this PR.
- [ ] If applicable, I have discussed the analytics needs at both a platform and study level with Product and instrumented code accordingly.
- [ ] If applicable, my UI/UX changes have passed muster with Product/Design via an over-the-shoulder review, screenshots, etc.

_If unsure or need help with any of the above items, add the `help wanted` label. For items that starts with `If applicable`, if it is not applicable, check it off and add `n/a` in front._

## FUD Score

_Overall, how are you feeling about these changes?_

- [x] :relaxed: All good, business as usual!
- [ ] :sweat_smile: There might be some issues here
- [ ] :scream: I'm sweaty and nervous

## How do we demo these changes?

_How does one observe these changes in a deployed system? Note that **user visible** encompasses many personas--not just patients and study staff, but also ops duty, your fellow devs, compliance, etc._

- [ ] They are user-visible in dev as a regular user journey and require no additional instructions.
- [ ] Getting dev into a state where this is user-visible requires some tech fiddling. I have documented these steps in the related ticket.
- [ ] Requires other features before it's human visible. I have documented the blocking issues in jira.
- [ ] I have no idea how to demo this. Please help me!

## Testing

- [ ] I have added regression test labels to the jira ticket
- [ ] I have added verification steps to the jira ticket
- [x] I have written automated positive tests
- [ ] I have written automated negative tests
- [ ] I have written zero automated tests but have poked around locally to verify proper functionality
- [ ] The jira ticket has acceptance criteria and QA has the needed information to test changes

## Release

- [x] These changes require no special release procedures--just code!
- [ ] Releasing these changes requires special handling and I have documented the special procedures in the release plan document

### CODE REVIEW

File: pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/service/adminoperation/ParticipantDataFixupService.java [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+
+    protected enum UpdateStatus {
+        UPDATED,
+        NOT_UPDATED,
+        ERROR,
+        NO_PARTICIPANT_DATA
+    }
+
+    private static final Gson gson = new Gson();
+    protected List<String> validRealms = List.of("atcp");
```
Comment (ssettipalli) [inline pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/service/adminoperation/ParticipantDataFixupService.java:39]: Just a thought.
If the service is specific to only ATCP, maybe the name should reflect it like ATParticipantDataFixupService.java
OR have ATParticipantDataFixupService.java as separate class extending ParticipantDataFixupService.java supporting only ATCP and have realm check in there.

**Review by ssettipalli — APPROVED** [review]

File: pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/service/adminoperation/ParticipantDataFixupService.java (hunk 1 above)
Comment (denniscunningham) [reply pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/service/adminoperation/ParticipantDataFixupService.java:39]: Yup, at this point it's hard to say what the next ParticipantData fixup would look like, and we may have no other cases, or we could have like 5.

What I tried to do here is strike a balance: The service is parameterized (at the HTTP request level) so we can use that to indicate the type of fixup, but of course there is only one now. I think if the next fixup is a lot like this one (uses much of the same code) then I would probably just keep the structure as is and put conditionals in the code. If the next case is different enough to warrant a separate class, then likely we would implement a class hierarchy as you describe, or perhaps just go with two separate (but renamed) classes.

I am working under the "you're not gonna need it (YAGNI)" principal, and I don't think this implementation has ruled out any future options.

Does that make sense? What do you think?

**Review by denniscunningham — COMMENTED** [review]
Thanks for your review @ssettipalli!
I left a response to you comment.

File: pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/service/adminoperation/ParticipantDataFixupService.java (hunk 1 above)
Comment (ssettipalli) [reply pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/service/adminoperation/ParticipantDataFixupService.java:39]: sounds ok to me.

**Pull request merged**